addons/stock/tests/test_report.py

```diff
@@ -1823,3 +1824,46 @@ def test_report_reception_7_done_receipt(self):
         self.assertEqual(mto_move.product_uom_qty, incoming_qty, "Move quantities should be unchanged")
         self.assertEqual(mto_move.procure_method, 'make_to_stock', "Procure method not correctly reset")
         self.assertEqual(mto_move.state, 'assigned', "Unassigning receipt move shouldn't affect the out move reservation")
+
+    def test_report_reception_immediate_transfer(self):
+        """ Having a sale order, a receipt with a move line created before the move
+        (immediate transfer) for the product of the SO should have its `Allocation` option entries
+        correctly populated with the sale order.
+        """
+        planned_delivery = self.env['stock.picking'].create({
+            'picking_type_id': self.picking_type_out.id,
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.customer_location.id,
+            'move_ids': [(0, 0, {
+                'name': 'TAFIT move',
+                'location_id': self.stock_location.id,
+                'location_dest_id': self.customer_location.id,
+                'product_id': self.product.id,
+                'product_uom_qty': 1,
+            })]
+        })
+        planned_delivery.action_confirm()
+
+        immediate_receipt_transfer = self.env['stock.picking'].create({
+            'picking_type_id': self.picking_type_in.id,
+            'location_id': self.supplier_location.id,
+            'location_dest_id': self.stock_location.id,
+            'move_line_ids': [(0, 0, {
+                'location_id': self.supplier_location.id,
+                'location_dest_id': self.stock_location.id,
+                'product_id': self.product.id,
+                'quantity': 1,
+            })],
+        })
+        immediate_receipt_transfer.button_validate()
+
+        allocation_entries = self.env['report.stock.report_reception'].get_report_data(
+            immediate_receipt_transfer.ids, {
+                'context': {
+                    'active_id': immediate_receipt_transfer.id, 'active_ids': immediate_receipt_transfer.ids,
+                    'active_model': 'stock.picking', 'default_picking_ids': immediate_receipt_transfer.ids,
+                }
+            }
+        )
```

Comment:
Instead of all of this, you can just use `self.env['report.stock.report_reception']._get_report_values(docids=[immediate_receipt_transfer.id])`

Also, checking that the picking name is in there is ok, but you should also check the qty to make sure the math was done correctly. It might be worth doing the picking uom != to the product uom and checking that it's correctly converted in the reception report values